### ISSUE TITLE
ARM64 docker image: no sound

### ISSUE THREAD

**Issue by nmaster2042** [jketterl/openwebrx#41]
I'm using a Khadas vim2 SBC with amlogic S912 processor with docker.

Contrary to tests made with my main computer ubuntu 19.10 x86_64 docker, on the SBC I got no sound whatever modulation I chose, whereas everything seems to be ok (no errors). 

The only thing strange is in status, CPU usage stays at 0%.

Both PC and SBC are on the same local lan.

Here is my logs from openwebrx docker container:

root@Khadas:~/docker/images/openwebrx# docker-compose up
Starting openwebrx ... 
Starting openwebrx ... done
Attaching to openwebrx
openwebrx    | 2019-12-30 14:35:44,181 - owrx.sdr - INFO - SDR sources loaded. Availables SDRs: Airspy
openwebrx    | 2019-12-30 14:35:56,568 - owrx.connection - DEBUG - client connection intitialized
openwebrx    | 2019-12-30 14:35:56,594 - owrx.source - DEBUG - activating profile VHF
openwebrx    | 2019-12-30 14:35:56,623 - owrx.source - INFO - Started sdr source: soapy_connector -s 2500000 -P 0 -g 10 -d driver=airspy -f 150000000 -p 4950 -c 42825
openwebrx    | [ERROR] SoapySDR::loadModule(/usr/local/lib64/SoapySDR/modules0.8/libsdrPlaySupport.so)
openwebrx    |   dlopen() failed: Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by /usr/local/lib/libmirsdrapi-rsp.so.2.13)
openwebrx    | Tuner gain set to 10.00 dB.
openwebrx    | IQ worker thread started
openwebrx    | setting up control socket...
openwebrx    | socket setup complete, waiting for connections
openwebrx    | control socket started on 42825
openwebrx    | [INFO] Using format CS16.
openwebrx    | 2019-12-30 14:35:56,728 - owrx.source.connector - DEBUG - opening control socket...
openwebrx    | client connection establised
openwebrx    | 2019-12-30 14:35:56,730 - owrx.dsp - DEBUG - received STATE_RUNNING, attempting DspSource restart
openwebrx    | 2019-12-30 14:35:56,732 - csdr.csdr - DEBUG - Command = nc -v 127.0.0.1 4950 | csdr shift_addition_cc --fifo /tmp/openwebrx/openwebrx_pipe_548307058384_shift_pipe | csdr fir_decimate_cc 226 0.000663716814159292 HAMMING | csdr bandpass_fir_fft_cc --fifo /tmp/openwebrx/openwebrx_pipe_548307058384_bpf_pipe 0.028928000000000002 HAMMING | csdr squelch_and_smeter_cc --fifo /tmp/openwebrx/openwebrx_pipe_548307058384_squelch_pipe --outfifo /tmp/openwebrx/openwebrx_pipe_548307058384_smeter_pipe 5 1 | csdr fmdemod_quadri_cf | csdr limit_ff | csdr fractional_decimator_ff 1.0033511929845684 | csdr deemphasis_nfm_ff 11025 | csdr convert_f_s16 | csdr encode_ima_adpcm_i16_u8
openwebrx    | control connection established
openwebrx    | 2019-12-30 14:35:56,757 - owrx.dsp - DEBUG - adding new output of type audio
openwebrx    | Connection to 127.0.0.1 4950 port [tcp/*] succeeded!
openwebrx    | client connection establised
openwebrx    | csdr bandpass_fir_fft_cc: fifo control mode on
openwebrx    | fir_decimate_cc: taps_length = 6027
openwebrx    | csdr fractional_decimator_ff: use_prefilter = 0, num_poly_points = 12, transition_bw = 0.03, window = HAMMING
openwebrx    | csdr fractional_decimator_ff: not using taps
openwebrx    | closing client socket
openwebrx    | csdr shift_addition_cc: fifo control mode on
openwebrx    | csdr squelch_and_smeter_cc: fifo control mode on
openwebrx    | csdr bandpass_fir_fft_cc: (fft_size = 512) = (taps_length = 139) + (input_size = 374) - 1
openwebrx    | (overlap_length = 138) = taps_length - 1
openwebrx    | csdr shift_addition_cc: reinitialized to -0
openwebrx    | csdr squelch_and_smeter_cc: initial squelch level is 1e-15
openwebrx    | 2019-12-30 14:35:56,781 - owrx.dsp - DEBUG - adding new output of type smeter
openwebrx    | 2019-12-30 14:35:56,792 - owrx.bands - WARNING - Frequency for wspr on 4m is not within band limits: 70091000
openwebrx    | 2019-12-30 14:35:56,805 - owrx.fft - DEBUG - Spectrum thread initialized successfully.
openwebrx    | 2019-12-30 14:35:56,806 - csdr.csdr - DEBUG - Command = nc -v 127.0.0.1 4950 | csdr fft_cc 4096 2863.68843069874 | csdr logaveragepower_cf -70 4096 97 | csdr fft_exchange_sides_ff 4096 | csdr compress_fft_adpcm_f_u8 4096
openwebrx    | csdr bandpass_fir_fft_cc: filter initialized, low_cut = -0.3616, high_cut = 0.3616
openwebrx    | Connection to 127.0.0.1 4950 port [tcp/*] succeeded!
openwebrx    | client connection establised

**Comment by jketterl** [comment]
Well, the log shows no real problems (apart from the sdrplay one which i cannot fix, unfortunately). Are you getting an S-Meter readout, or waterfall? Can you try moving the squelch? (I've had the squelch stuck in closed for some times now, no idea why it happens, but moving it all the way across usually solves the problem).

I don't think it's the FFTW_MEASURE bug since the libs in the container should not be affected, but it's not explicitly disabled, sojust in case: please keep the receiver window open and wait for ~30 seconds. If there's no audio then, it's not that one.

**Comment by nmaster2042** [comment]
Yes I have the waterfall and smeter reading see below.

![Screenshot_20191230_164156](https://user-images.githubusercontent.com/22748840/71588935-8b59f680-2b23-11ea-8096-19e73dd513ef.png)

Tried moving squelch with no luck for now.

Letting received for more than 30 second is the same: no sound.

**Comment by nmaster2042** [comment]
I also tried with another browser Chromium, same result.

If I move volume button completely right (full) I get little noise but it doesn't change when I change modulation.

**Comment by jketterl** [comment]
when i look at the screenshot, i can see that your tuned frequency is out of band. not sure if you tried tuning after that, but please try to do so if you haven't. also, try setting the `start_freq` in your profile to something in-range.

**Comment by nmaster2042** [comment]
I made a lot of tests, so I may be wrong on some of them.

I made new tests, here is my sdr configuration:

sdrs = {
    "airspy": {
        "name": "Airspy",
        "type": "airspy",
        "ppm": 0,
        "profiles": {
            "VHF": {
                "name": "VHF",
                "center_freq": 146000000,
                "rf_gain": 10,
                "samp_rate": 2500000,
                "start_freq": 145800000,
                "start_mod": "nfm",
            },
        },
    },
}

And the corresponding screenshot:

![Screenshot_20191231_175834](https://user-images.githubusercontent.com/22748840/71628281-7fd7ff80-2bf7-11ea-83a0-8fbbf7c9eb53.png)

Still no sound.

**Comment by jketterl** [comment]
i'm out of ideas for the moment. as i said, i've had very quiet audio before, and usually it goes away when you move the squelch, but as i said, i don't know what's causing it. i will get back to this when i find out more.

**Comment by nmaster2042** [comment]
Happy new year.

I just tried the docker image on another arm64 SBC: an odroid C2 with an amlogic S905x processor running the lastest armbian version.

With this setup, all is fine and I get sound.

It was for test purpose only because this SBC has less power and can't handle the default 10 Mbps sample rate of my airpsy contrary to my khadas.

I'm not a docker expert, the only software difference is the OS. On my khadas VIM2, it's ubuntu bionic witch is older than armbian.

On hardware side the difference is an amlogic S905x 4 cores processor against an amlogic S912 8 cores processor on khadas vim2.

Could it be a build flag on some software part of the image ?
Or something else, I don't know.

But we can see amr64 docker image is working.

**Comment by nmaster2042** [comment]
Hi.

I found an newer os image for my SBC, armbian for khadas SBC and other amlogic processors.

It's a work in progress and not official armbian supported but it has newer kernel and base os components.

I burned it on a micro sdcard and installed docker on it.

And I have audio now.

So my issue doesn't seem to be directly related to openwebrx image.

I wanted to let you know, it may be useful for other users.
